tools/mo/openvino/tools/mo/ops/parameter.py

```diff
@@ -56,6 +56,11 @@ def reverse_infer(node: Node):
         # update node 'shape' attribute (if it is not defined) from the output port shape which was calculated
         # during the reverse_infer phase
         shape = node.soft_get('shape', None)
-        if shape is None and node.out_port(0).data.get_shape() is not None:
-            node['shape'] = node.out_port(0).data.get_shape()
-
+        if shape is not None:
+            return
+        for i, out_port in node.out_ports().items():
+            out_shape = out_port.data.get_shape()
+            # todo: add comparison and choose most defined partial shape,
```

Comment:
Do we need to implement this logic in this PR or it is a part of future work?

Reply:
I didn't have intention to do it this PR, because we anyway at the moment get fully dynamic shapes with only known rank, like `[-1, -1, -1, -1]`.

But I left this comment for the future if we will reversely propagate also partially defined shapes like `[-1, 3, -1, -1]`.

Reply:
> BTW looks good. Do we have some real models to check these changes?

I manually checked on a bunch of models:
face-recognition-resnet34-arcface
mobilefacedet-v1-mxnet
octave-resnext-101-0.25
mobilenet-v2-1.0-224
mobilenet-v2-1.4-224
dense_vnet_abdominal_ct
resnet-50-tf
deepvariant_wgs
ResNext101-64x64-mxnet
and bunch of others
and many others

but didn't automatically run. Today I run locally frameworks.ai.openvino.tests on a full list of e2e models. So far results are green. Will update when it's finished

Reply:
Anyway, i implemented this logic in the current PR. I check number of dynamic dimensions by `np.ma.count_masked`. 

If Parameter had several outputs and `reverse_infer` deduced shapes not for all of them this logic will choose the best defined one.